cmd/agent/main.go

```diff
@@ -162,7 +164,7 @@ func startAPIServer(config *http.APIServerConfig) error {
 	return server.StartSecured()
 }
 
-func enableEdgeMode(tunnelOperator agent.TunnelOperator, clusterService agent.ClusterService, options *agent.Options) error {
+func enableEdgeMode(tunnelOperator agent.TunnelOperator, clusterService agent.ClusterService, infoService agent.InfoService, options *agent.Options) error {
```

Comment:
The `enableEdgeMode` function starts to look complex/hard to read, I would advise to extract it in multiple functions.

Here is a proposition of the final result for that function (related functions not provided):

```go
func enableEdgeMode(tunnelOperator agent.TunnelOperator, clusterService agent.ClusterService, infoService agent.InfoService, options *agent.Options) error {
	edgeKey, err := retrieveEdgeKey(options, clusterService)
	if err != nil {
		return err
	}

	if edgeKey != "" {
		log.Println("[DEBUG] [main,edge] [message: Edge key available. Starting tunnel operator.]")

		err := associateEdgeKey(tunnelOperator, clusterService, edgeKey)
		if err != nil {
			return err
		}
	} else {
		log.Println("[DEBUG] [main,edge] [message: Edge key not specified. Serving Edge UI]")
		serveEdgeUI(tunnelOperator, clusterService, options)
	}

	err = startRuntimeConfigCheckProcess(tunnelOperator, infoService)
	if err != nil {
		return err
	}

	return nil
}
```